app/routes/admin/activities/add.js

```diff
@@ -115,6 +116,8 @@ module.exports = [
         stepThreeCheckConfirmation
       ],
       handler: async (request, h) => {
+        throwIfPreConditionError(request)
```

Comment:
For all these scenarios, is the throwIfPreConditionError() only needed on step 3 each time?